### DISCUSSION TITLE
求助 LightApp [mirai:app:$content] 无法发送

### DISCUSSION THREAD

**Discussion by cuicanontop** [simple-robot/Discussions#47] [Q&A]
LightApp lightApp = new LightApp("");
final Neko neko = CatCodeMessageUtil.toNeko(lightApp);
final Message.Element<?> message = CatCodeMessageUtil.toMessage(neko);
event.getGroup().sendBlocking(message);
请问是写错了还是卡片被屏蔽了？

**Comment by ForteScarlet** [comment]
猫猫码在3.x中已经被弃用，`CatCodeMessageUtil` 仅仅是mirai组件为了提供一定程度的**兼容**而存在的，如非为了临时兼容旧代码则不应再使用猫猫码。

在 3.x 中如果希望发送 `LightApp` ，可以参考如下代码：
```java
LightApp lightApp = new LightApp(...);
Message.Element<?> message = MiraiMessageParserUtil.asSimbotMessage(lightApp);
event.getGroup().sendBlocking(message);
```

其次，`LightApp` 本身就不稳定，发送不了也很正常
